### PR TITLE
Fix Vue pug rendering by specifying HTML5 by default

### LINKED ISSUE #7756: Vue 3: "v-else" conditional compilation error in Pug template (SFC)
# 🐛 bug report

Compile a simple Vue 3 app with Parcel 2 (watch mode) and use a `v-if` and `v-else` conditional statement in a Pug template, Vue Transformer will output a parse error.

Single File Component Example:
```pug
<template lang="pug">

.wrapper

	p(v-if="false") foo
	p(v-else) bug

</template>
```

Error:
```bash
@parcel/transformer-vue: Error parsing JavaScript expression: Unexpected token (1:3)
````

Also I noticed, using instead `v-else=""` the template compiles successfully.

## 🎛 Configuration (.babelrc, package.json, cli command)

```json
{
	"name": "vue-parcel",
	"description": "Simple boilerplate for building static SPA's with VueJs 3.",
	"license": "UNLICENSED",
	"main": "app/app.js",
	"repository": "https://github.com/npulidom/vue-parcel",
	"scripts": {
		"watch": "rm -rf dist && parcel app/index.pug --dist-dir dist --cache-dir .parcel --no-source-maps",
		"build": "rm -rf dist && parcel build app/index.pug --dist-dir dist --cache-dir .parcel --no-source-maps --detailed-report 20",
		"clean": "rm -rf dist .parcel"
	},
	"browserslist": "> 0.5%, last 2 versions, not dead",
	"devDependencies": {
		"@parcel/config-default": "~2.3.2",
		"@parcel/core": "~2.3.2",
		"@parcel/optimizer-cssnano": "~2.3.2",
		"@parcel/transformer-pug": "~2.3.2",
		"@parcel/transformer-stylus": "~2.3.2",
		"@parcel/transformer-vue": "~2.3.2",
		"@vue/component-compiler-utils": "~3.3.0",
		"autoprefixer": "~10.4.2",
		"cssnano": "~5.0.17",
		"parcel-resolver-vue3-runtime": "~2.0.2",
		"postcss": "~8.4.6",
		"pug": "~3.0.2",
		"stylus": "~0.57.0"
	},
	"dependencies": {
		"modern-normalize": "~1.1.0",
		"vue": "~3.2.31",
		"vue-router": "~4.0.12",
		"vuex": "~4.0.2"
	}
}
```

## 🤔 Expected Behavior

Vue Transformer should support `v-else` in Pug templates.

## 😯 Current Behavior

Compilation Error

```bash
🚨 Build failed.

@parcel/transformer-vue: Error parsing JavaScript expression: Unexpected token (1:3)

SyntaxError: Error parsing JavaScript expression: Unexpected token (1:3)
```

## 💁 Possible Solution

Check `v-else` statement in pug template of SFCs. 

## 🔦 Context

Compile a simple Vue 3 application using a Pug template in a SFC.

## 💻 Code Sample

Simple reproduction [here](https://github.com/npulidom/vue-parcel/tree/issue-pug-v-else).

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           |  2.3.2
| Node             |  14.18.1
| npm/Yarn         | yarn 1.22.17
| Operating System |  Manjaro Linux

### PR DESCRIPTION
**Pull request by Jengamon**
# ↪️ Pull Request

For the pug template engine for Vue, make HTML5 the default. This will fix #7756.
See discussion at #8050 for considerations about this change.

## 💻 Examples

```vue
<template lang="pug">

.wrapper

	p(v-if="false") foo
	p(v-else) bug

</template>
```
should now render properly without any workarounds.

## 🚨 Test instructions

I will add the above example as an integration test.

## ✔️ PR Todo

- [ ] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

**Comment by Jengamon** [comment]
Does this need a whole integration test?

File: packages/transformers/vue/src/VueTransformer.js [hunk 1]
```diff
@@ -225,7 +225,16 @@ async function processPipeline({
...
       if (template.lang && !['htm', 'html'].includes(template.lang)) {
+        let options = {};
         let preprocessor = consolidate[template.lang];
+        // Pug doctype fix (fixes #7756)
+        switch (template.lang) {
+          case 'pug':
+            options.doctype = 'html';
+            break;
+          default:
+            break;
```
Comment (101arrowz) [inline packages/transformers/vue/src/VueTransformer.js]: Not a big issue but these lines aren't really necessary.

**Review by 101arrowz — APPROVED** [review]
LGTM

File: packages/transformers/vue/src/VueTransformer.js (hunk 1 above)
Comment (Jengamon) [reply packages/transformers/vue/src/VueTransformer.js]: True, I can remove them

**Review by devongovett — APPROVED** [review]

**Pull request merged**